Cargo.toml

```diff
@@ -51,13 +51,19 @@ regex = "1.3.1"
 os_info = "2.0.1"
 urlencoding = "1.0.0"
 open = "1.4.0"
-# OpenSSL causes problems when building a MUSL release. Opting to use native SSL implementation
-# see: https://github.com/richfelker/musl-cross-make/issues/65#issuecomment-509790889
-reqwest = { version = "0.10.4", default-features = false, optional = true, features = ["blocking", "rustls-tls"] }
 unicode-width = "0.1.7"
 textwrap = "0.11.0"
 term_size = "0.3.1"
 
+# OpenSSL causes problems when building a MUSL release.
```

Comment:
Oof that's hairy. I wish there was a more concise way to address this. 😰

Reply:
In #933 @q66 mentions everything builds fine for them with MUSL. Is the trouble related with MUSL and openssl related to the CI setup or is it something else?

Should I start a test run to see if MUSL builds fine with openssl on CI now? Beyond that do you know how well the rust openssl crate handles different openssl library versions and  locations across different systems these days? I remember having trouble with that back in the day.

Reply:
yes, no issues building on musl (x86_64 or any other architecture) on my system when using `openssl-sys`, it just dynamically links against that and works fine

i think the issue for you might be that upstream rust musl builds prefer static linkage by default, while we override this in Void to use dynamic linkage by default; that may result in the openssl crate building its own openssl rather than linking system's...

so, maybe make the condition something like `cfg(target_feature = "crt-static")`? if using `crt-static`, use `rustls`, otherwise use `native-tls`... then see if it builds on CI, and I can test if it builds here.

Reply:
> i think the issue for you might be that upstream rust musl builds prefer static linkage by default, while we override this in Void to use dynamic linkage by default; that may result in the openssl crate building its own openssl rather than linking system's...

Well I just tried cross-compiling to FreeBSD and I just the same error as when I tried to compile to MUSL (`Could not find directory of OpenSSL installation` which apparently means it needs a cross-compiled openssl). This means that the error is likely not related to `crt-static` (and isn't that anly about the c runtime and not other dependencies?). I suppose vendoring openssl could be a fix as opposed it to being the issue.

> Ideally, it'd be good to make this overridable too, to force either of these options even if the default is different.

Not sure how this could be solved. Is it possible to do feature-aware dependency specification?

Reply:
Yes, if you cross compile you obviously need OpenSSL/libressl for target in the cross root, that works fine though (we build all ARM packages as crosscompiled in Void and there are no issues)

Reply:
I think the MUSL builds here are mainly intended for maximum platform compablity so I guess vendoring works fine here.

Reply:
It makes sense as an option, i don't think musl is otherwise special though

Reply:
Yes. I added a disabled-by-default `tls-vendored` feature in the recent commits that gets enabled only in the CI MUSL release builds.